src/native/eventpipe/ep.c

```diff
@@ -571,11 +571,84 @@ disable_helper (EventPipeSessionID id)
 		EventPipeProviderCallbackDataQueue callback_data_queue;
 		EventPipeProviderCallbackData provider_callback_data;
 		EventPipeProviderCallbackDataQueue *provider_callback_data_queue = ep_provider_callback_data_queue_init (&callback_data_queue);
+		EventPipeSession *session = NULL;
+		EventPipeThread * thread = NULL;
 
-		EP_LOCK_ENTER (section1)
-			if (ep_volatile_load_number_of_sessions () > 0)
-				disable_holding_lock (id, provider_callback_data_queue);
```

Comment:
Is disable_holding_lock called anywhere else? It looks like you duplicated the code here, should we delete that method now?